src/index.js

```diff
@@ -22,11 +22,11 @@ export var GestureHandling = L.Handler.extend({
 		this._disableInteractions();
 
 		//Uses native event listeners instead of L.DomEvent due to issues with Android touch events turning into pointer events
-		this._map._container.addEventListener("touchstart", this._handleTouch);
-		this._map._container.addEventListener("touchmove", this._handleTouch);
-		this._map._container.addEventListener("touchend", this._handleTouch);
-		this._map._container.addEventListener("touchcancel", this._handleTouch);
-		this._map._container.addEventListener("click", this._handleTouch);
+		this._map._container.addEventListener("touchstart", this._handleTouch, { passive: true });
+		this._map._container.addEventListener("touchmove", this._handleTouch, { passive: true });
+		this._map._container.addEventListener("touchend", this._handleTouch, { passive: true });
+		this._map._container.addEventListener("touchcancel", this._handleTouch, { passive: true });
+		this._map._container.addEventListener("click", this._handleTouch, { passive: true });
```

Comment:
Hi @bertho-zero,

thanks for the contribution, but are you sure it's working properly?

## • [docs/addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener)

> **passive**
>
> A boolean value that, if true, indicates that the function specified by listener will never call [preventDefault()](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault). If a passive listener does call preventDefault(), the user agent will do nothing other than generate a console warning. If not specified, defaults to false – except that in browsers other than Safari and Internet Explorer, defaults to true for the [wheel](https://developer.mozilla.org/en-US/docs/Web/API/Element/wheel_event), [mousewheel](https://developer.mozilla.org/en-US/docs/Web/API/Element/mousewheel_event), [touchstart](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchstart_event) and [touchmove](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchmove_event) events. See [Improving scrolling performance with passive listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#improving_scrolling_performance_with_passive_listeners) to learn more.

I haven't looked at this library in a while, but taking a quick look at the code it would seem to be just the opposite of the above, eg:

https://github.com/Raruto/leaflet-gesture-handling/blob/f31da5ebff8bca77cb06466f36c16c5b0d4e0034/src/index.js#L199

The opposite value seems to be implemented in leaflet core as well:

## • [leaflet/src/dom/DomEvent.js#L120-L121](https://github.com/Leaflet/Leaflet/blob/8c38dbafb614a887a546e64aff913dc7feda9a0d/src/dom/DomEvent.js#L120-L121)
```js
if (type === 'touchstart' || type === 'touchmove' || type === 'wheel' ||  type === 'mousewheel') {
  obj.addEventListener(mouseSubst[type] || type, handler, {passive: false});
```

---

**PS** should it be set on the click as well?

> this._map._container.addEventListener("click", this._handleTouch, { passive: true });

Reply:
> I tested on a map I have in production and I don't see any difference neither on mobile nor on browser, everything works perfectly.

Please provide some additional information on the tests performed (browser, device, checks performed, ...)

I don't want to sound picky, but I know that until now it worked somehow on mobile, and since I don't have a touch device 🤦‍♂️ I would prefer to make certain changes too much quickly (and just to please big G).

Also, for some inspiration for your tests, please take a look at the:

- [leaflet-gesture-handling/test/index.html](https://raruto.github.io/leaflet-gesture-handling/test/index.html) page 

- older issues from [elmarquis/Leaflet.GestureHandling](https://github.com/elmarquis/Leaflet.GestureHandling/issues?q=)

If I remember correctly the main problems could be created when interacting with markers, popups and controls under the warning overlay.


**BTW**, since I think that the leaflet core library already handles those options (in a compatible manner with lighthouse tests), also try to see if by updating leaflet library to a more recent version we can switch back writing those listener through `L.DomEvent` calls:

```js
addHooks: function() {

  // Uses native event listeners instead of L.DomEvent due to issues with Android touch events turning into pointer events

  // this._map._container.addEventListener("touchstart", this._handleTouch, { passive: true });
  // this._map._container.addEventListener("touchmove", this._handleTouch, { passive: true });
  // this._map._container.addEventListener("touchend", this._handleTouch, { passive: true });
  // this._map._container.addEventListener("touchcancel", this._handleTouch, { passive: true });
  // this._map._container.addEventListener("click", this._handleTouch, { passive: true });

  L.DomEvent.on(this._map._container, "touchstart", this._handleTouch, this);
  L.DomEvent.on(this._map._container, "touchmove", this._handleTouch, this);
  L.DomEvent.on(this._map._container, "touchend", this._handleTouch, this);
  L.DomEvent.on(this._map._container, "touchcancel", this._handleTouch, this);
  L.DomEvent.on(this._map._container, "click", this._handleTouch, this);

...
}

removeHooks: function() {

  // Uses native event listeners instead of L.DomEvent due to issues with Android touch events turning into pointer events

  // this._map._container.removeEventListener("touchstart", this._handleTouch);
  // this._map._container.removeEventListener("touchmove", this._handleTouch);
  // this._map._container.removeEventListener("touchend", this._handleTouch);
  // this._map._container.removeEventListener("touchcancel", this._handleTouch);
  // this._map._container.removeEventListener("click", this._handleTouch);

  L.DomEvent.off(this._map._container, "touchstart", this._handleTouch, this);
  L.DomEvent.off(this._map._container, "touchmove", this._handleTouch, this);
  L.DomEvent.off(this._map._container, "touchend", this._handleTouch, this);
  L.DomEvent.off(this._map._container, "touchcancel", this._handleTouch, this);
  L.DomEvent.off(this._map._container, "click", this._handleTouch, this);

...
}
```

_Thanks again,
I look forward to your updates_

👊 Raruto

Reply:
I tested on Brave mobile (Android) and Brave linux with the latest version of leaflet/react-leaflet (1.9.3/2.1.0):
- the warning is displayed correctly
- I can move/zoom with two fingers
- the interactive parts work correctly on click/touch without displaying the warning (marker, zoom, etc.)

I test by trying to use `L.DomEvent.on` and tell you

Reply:
I tested with `L.DomEvent.on`/`L.DomEvent.off`, everything works fine on Brave + Chrome (mobile + pc) and Lighthouse does not complain